### PR TITLE
Added clearMenuIndices

### PR DESCRIPTION
**Pull request by tsitsop**
Menu indices are now cleared upon the closing of certain menus.

### CODE REVIEW

File: src/MenuGraphr.ts [hunk 1]
```diff
@@ -290,6 +290,9 @@ export class MenuGraphr implements IMenuGraphr {
         const menu: IMenu = this.menus[name];
 
         if (menu) {
+            if (menu.clearedIndicesOnDeletion) {
+                this.clearMenuIndices(name);
+            }
```
Comment (JoshuaKGoldberg) [inline src/MenuGraphr.ts]: Nit: try to keep a blank line after `}`s

File: src/MenuGraphr.ts [hunk 2]
```diff
@@ -721,6 +724,19 @@ export class MenuGraphr implements IMenuGraphr {
...
 
     /**
+     * Clears saved indices that should be forgotten when the menu is deleted.
+     * 
+     * @param name   The name of the menu that is being deleted.
+     */
+    public clearMenuIndices(name: string): void {
+        const menu: IMenu = this.menus[name];
+
+        for (let i: number = 0; i < menu.clearedIndicesOnDeletion.length; i += 1) {
```
Comment (JoshuaKGoldberg) [inline src/MenuGraphr.ts]: This would be slightly simpler with a `for..of` loop.

File: src/MenuGraphr.ts [hunk 3]
```diff
@@ -721,6 +724,19 @@ export class MenuGraphr implements IMenuGraphr {
     }
 
     /**
+     * Clears saved indices that should be forgotten when the menu is deleted.
+     * 
+     * @param name   The name of the menu that is being deleted.
+     */
+    public clearMenuIndices(name: string): void {
+        const menu: IMenu = this.menus[name];
```
Comment (JoshuaKGoldberg) [inline src/MenuGraphr.ts]: Use `this.getExistingMenu(name)` for public functions. Trying to clear indices of an unknown menu will then throw a better error.

Comment (tsitsop) [inline src/MenuGraphr.ts]: Do you want this function to be public? As of now it is only referenced in MenuGraphr, I didn't think about making it private

Comment (JoshuaKGoldberg) [inline src/MenuGraphr.ts]: Sure, let's make it private.

File: src/MenuGraphr.ts (hunk 1 above)
Comment (tsitsop) [inline src/MenuGraphr.ts]: This is fixed in the PR

**Pull request merged**